### ISSUE TITLE
Protect users from accidental cloud-config opt-in

### ISSUE THREAD

**Issue by gberche-orange** [cloudfoundry/bosh#1170]
According to http://bosh.io/docs/cloud-config.html, the cloud-config opt-in is global to the bosh director and can't be reverted unless editing the bosh db. 

The impact of an accidental opt-in seems that all existing deployments on the bosh-director need to be "cloud-config" friendly, i.e. not include the Iaas settings (network, compilation...). It seems that the main bosh releases such as cf-release (in [generate_deployment_manifest](https://github.com/cloudfoundry/cf-release/blob/master/scripts/generate_deployment_manifest#L25-L39) that [still require iaas-specific infras](https://github.com/cloudfoundry/cf-release/blob/master/templates/cf-infrastructure-vsphere.yml#L9-L81) )  have their default manifest generation not yet "cloud-config" friendly.  

As a result, an accidental optin on a bosh director production platform has the side effect that emergency upgrades (stemcells or other release upgrades for security fix) can't be applied.

See example of such accidental opt-in in https://cloudfoundry.slack.com/archives/bosh-users/p1458080218000039

Improvement suggestions:
- add a confirmation prompt on the cloud-config opt-in i.e. the first call to "bosh update cloud-config" that reminds the impact of the opt-in
- document the manual way to opt-out (clear the `cloud_configs` table and `deployments.cloud_config_id` column)
- introduce a bosh CLI command to opt-out, possibly in simple cases just after opting-in

**Comment by cppforlife** [comment]
That's fair. I'll try to write up a few steps to clean up the DB in emergency case or if you have a chance I would be happy to take a PR to docs-bosh with the steps you did. Im assuming you did them via `director_ctl console`?

**Comment by gberche-orange** [comment]
Thanks @cppforlife for your response. A documented opt-out roll back steps would certainly help. 

In our case we rolled back using regular SQL client. I was not familiar enough with the ``director_ctl console`` script to use it right away. Is it close to the CC console behavior (for which I had contributed [initial pointers ](https://github.com/cloudfoundry/docs-running-cf/commit/cc46acc26eff50988a652acc3582c7573d6c8c8e)) ?
